### PR TITLE
Highlight commit hook way of signoff

### PR DESCRIPTION
**Pull request by jpienaar**
This is a local to checkout change folks can follow.

### CODE REVIEW

File: docs/website/docs/developers/general/contributing.md [hunk 1]
```diff
@@ -65,6 +65,9 @@ Signed-off-by: Random J Developer <random@developer.example.org>
     git commit -s -m 'This is my commit message'
     ```
 
+  See `.git/hooks/prepare-commit-msg.sample` for an example of automaticall
+  adding as part of hook to prepare commit message.
```
Comment (ScottTodd) [inline docs/website/docs/developers/general/contributing.md:69]: Fix typo + some rewording suggestions

```suggestion
    See `.git/hooks/prepare-commit-msg.sample` for an example of automatically
    adding this using a [git hook](https://git-scm.com/docs/githooks) to prepare
    commit messages.
```

Also, my local file has
```
# SOB=$(git var GIT_COMMITTER_IDENT | sed -n 's/^\(.*>\).*$/Signed-off-by: \1/p')
# git interpret-trailers --in-place --trailer "$SOB" "$COMMIT_MSG_FILE"
# if test -z "$COMMIT_SOURCE"
# then
#   /usr/bin/perl -i.bak -pe 'print "\n" if !$first_line++' "$COMMIT_MSG_FILE"
# fi
```

but I might also try https://stackoverflow.com/a/46536244

**Review by ScottTodd — APPROVED** [review]

File: docs/website/docs/developers/general/contributing.md (hunk 1 above)
Comment (ScottTodd) [reply docs/website/docs/developers/general/contributing.md:69]: Tested the perl version in the local file, works on my Windows machine from CLI and `git gui`.

I'll also try commit signing 
* https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits
* https://docs.github.com/en/authentication/managing-commit-signature-verification/generating-a-new-gpg-key
* https://docs.github.com/en/authentication/managing-commit-signature-verification/associating-an-email-with-your-gpg-key
* https://docs.github.com/en/authentication/managing-commit-signature-verification/telling-git-about-your-signing-key

Comment (ScottTodd) [reply docs/website/docs/developers/general/contributing.md:69]: Commit signing works for me (after some OS-specific setup). The GitHub docs are pretty comprehensive, but we can condense them for common scenarios.

**Review by ScottTodd — CHANGES_REQUESTED** [review]
I incorporated this change into https://github.com/iree-org/iree/pull/17412

**Comment by ScottTodd** [comment]
My PR landed, commit hook is in the docs now

**Comment by jpienaar** [comment]
Thanks!

**Pull request closed without merging**